osbrain/tests/test_proxy.py

```diff
@@ -119,6 +122,41 @@ def ping(self):
         InitTimeoutProxy('foo', timeout=1.)
 
 
+def test_agent_proxy_wait_running(nsproxy):
+    """
+    Using wait_for_running on a proxy after initialization should block until
+    the agent is running or time out.
+    """
+    ap = AgentProcess('agent')
+    ap.start()
+
+    # Get "offline" proxy
+    agent = Proxy('agent')
+    Timer(.5, agent.run).start()
+
+    # Wait long enough for the agent to start running
+    proxy = Proxy('agent', timeout=1).wait_for_running()
+    assert proxy.ping() == 'pong'
+
+
+def test_agent_proxy_wait_running_timeout(nsproxy):
+    """
+    Using wait_for_running on a proxy after initialization should block until
+    the agent is running or time out.
+    """
+    ap = AgentProcess('agent')
+    ap.start()
+
+    # Get "offline" proxy
+    Proxy('agent', nsproxy.addr())
+
+    # Wait for too long before starting the agent
+    with pytest.raises(TimeoutError) as error:
+        Proxy('agent', nsproxy.addr()).wait_for_running(timeout=.5)
+
+    assert "Timed out" in str(error.value)
```

Comment:
Just if you want: you could assert the *approximate* time passed between the `Proxy()` call and the raised exception.

In example: for 0 seconds timeout check that less than 1 second passed. For 1 seconds timeout check that more than 1 seconds passed but less than 2 seconds.

Just to make the test a bit more complete. :blush: